src/interpolate.jl

```diff
@@ -40,6 +40,113 @@ function cell2node(cell_centered_data::AbstractArray{Float64})
+                               coordinates::AbstractArray{Float64},
+                               levels::AbstractArray{Int}, length_level_0::Float64,
+                               center_level_0::AbstractArray{Float64},
+                               slice_axis, slice_axis_intersect)
+
+  dimensions = Dict(
+    :x => 1,
+    :y => 2,
+    :z => 3
+  )
+  if !haskey(dimensions, slice_axis)
+    supported_dims = keys(dimensions)
+    error("illegal dimension '$slice_axis', supported dimensions are $supported_dims")
+  end
+  slice_axis_dimension = dimensions[slice_axis]
+  other_dimensions = [1, 2, 3][1:end .!= slice_axis_dimension]
+
+  # Extract data shape information
+  n_nodes_in, _, _, n_elements, n_variables = size(unstructured_data)
+
+  # Get node coordinates for DG locations on reference element
+  nodes_in, _ = gauss_lobatto_nodes_weights(n_nodes_in)
+
+  # New unstructured data has one dimension less.
+  # The redundant element ids are removed later.
+  new_unstructured_data = similar(unstructured_data[1, ..])
+
+  # Declare new empty arrays to fill in new coordinates and levels
+  new_coordinates = Array{Float64}(undef, 2, 0)
+  new_levels = Array{eltype(levels)}(undef, 0)
+
+  # Counter for new element ids
+  new_id = 0
+
+  # Save vandermonde matrices in a Dict to prevent redundant generation
+  vandermonde_to_2d = Dict()
+
+  # Limits of domain in slice_axis dimension
+  lower_limit = center_level_0[slice_axis_dimension] - length_level_0 / 2
+  upper_limit = center_level_0[slice_axis_dimension] + length_level_0 / 2
+
+  if slice_axis_intersect < lower_limit || slice_axis_intersect > upper_limit
+    error("slice_axis_intersect $slice_axis_intersect outside of domain")
+  end
+
+  for element_id in 1:n_elements
+    # Distance from center to border of this element (half the length)
+    element_length = length_level_0 / 2^levels[element_id]
+    first_coordinate = coordinates[:, element_id] .- element_length / 2
+    last_coordinate = coordinates[:, element_id] .+ element_length / 2
+
+    # Check if slice plane and current element intersect
+    # The upper limit check is needed because of the > in the first check
+    if (first_coordinate[slice_axis_dimension] <= slice_axis_intersect &&
+          last_coordinate[slice_axis_dimension] > slice_axis_intersect) ||
+        (slice_axis_intersect == upper_limit &&
+          last_coordinate[slice_axis_dimension] == upper_limit)
+      # This element is of interest
+      new_id += 1
+
+      # Add element to new coordinates and levels
+      new_coordinates = hcat(new_coordinates, coordinates[other_dimensions, element_id])
+      push!(new_levels, levels[element_id])
+
+      # Construct vandermonde matrix (or load from Dict if possible)
+      normalized_intersect =
+          (slice_axis_intersect - first_coordinate[slice_axis_dimension]) /
+          element_length * 2 - 1
+
+      if haskey(vandermonde_to_2d, normalized_intersect)
```

Comment:
Instead of indexing by the normalized intersect, you could also just use the level, which might be faster (you could even store the vandermondes in a Vector of Arrays in that case)

Reply:
That would generate redundand vandermondes as there can be multiple levels using the same normalized intersect (and therefore the same vandermonde). This happens when a slice plane is at the edges of elements. For example consider the following example:
![grafik](https://user-images.githubusercontent.com/44124897/95855681-10cc3180-0d59-11eb-940a-b76c8eb78b24.png)
We want to cut at the red line and therefore interpolate for all elements touching the red line from the right. We want to interpolate all of them to their left border, so we use the same normalized intersect (-1) and the same vandermonde, although they all have different levels.

Reply:
OK, makes sense :+1: